lib/hooks.js

```diff
@@ -53,6 +53,44 @@ Hooks.prototype.validate = function (hook, fn) {
   }
 }
 
+Hooks.prototype.validateAsyncHookParams = function (hook, fn) {
+  if (fn.constructor.name !== 'AsyncFunction') {
+    return
+  }
+
+  let shouldThrow = false
+
+  switch (hook) {
```

Comment:
I'd add a comment above this switch to explain why the `fn.length` is being inspected.

Reply:
Will do, thanks!

Had a lot on my plate for the past few weeks, will get around to this during the next week.